#5506 - Create upvote popover

Miaplacidus opened the pull request:
This PR completes ticket #5506, which asks for the creation of a new upvote popover component. The upvote popover contains a list of users that have upvoted a thread. It also details the total number of voters and the voting weight of each individual voter.

## Link to Issue
Closes: #5506 

## Description of Changes
- creates component `UpvotePopover`
- adds `UpvotePopover` to components showcase

## Test Plan
- navigate to `/components`
- go to 'Upvote Popover' section and check that behavior of component is as expected

mhagel reviewed (approved):
Looks great. Left a comment to consider.

masvelio commented on packages/commonwealth/client/scripts/views/components/UpvotePopover/UpvotePopover.tsx:
```diff
@@ -0,0 +1,56 @@
+
+    if (upvoterAddresses.length > maxVisibleUpvotingAccounts) {
+      slicedUpvoters.push(
+        <div
```
let's use <CWText> with type `caption` to align styling with figma

Miaplacidus replied:
Done!

masvelio commented on packages/commonwealth/client/scripts/views/components/UpvotePopover/UpvotePopover.tsx:
```diff
@@ -0,0 +1,56 @@
+              userCommunityId={app.chain?.id || '1inch'}
+              shouldLinkProfile
+            />
+            <div className="vote-weight">5x</div>
```
use CWText

masvelio commented on packages/commonwealth/client/scripts/views/components/UpvotePopover/UpvotePopover.tsx:
```diff
@@ -0,0 +1,56 @@
+}: UpvotePopoverProps) => {
+  const maxVisibleUpvotingAccounts = 3;
+
+  const createUpvoterList = (upvoterAddresses) => {
```
type is missing for `upvoterAddresses`

Miaplacidus replied:
Done!

masvelio commented on packages/commonwealth/client/scripts/views/components/UpvotePopover/UpvotePopover.tsx:
```diff
@@ -0,0 +1,56 @@
+import './UpvotePopover.scss';
+
+type UpvotePopoverProps = {
+  upvoters: string[];
```
I think the default value `upvoters = []` would do the job, but below in line 14, instead of here in type

Miaplacidus replied:
Default value  of `[]` set. Thanks!

masvelio reviewed (changes requested):
Left some comments and also spotted some things that needs to be improved. 

1. I am not able to move my mouse over the popover eg if I want to click the user.

https://github.com/hicommonwealth/commonwealth/assets/14819225/4a083667-c0a4-46f0-b0ec-c09e27799b59

2. If upvoters array is empty, we display popover with only title. We should not display it at all as we do it currently in upvote button. The problem is that currently we render two different things depending if we have upvoters or not. We should do it better now and handle this logic within the component. 

![image](https://github.com/hicommonwealth/commonwealth/assets/14819225/5efb7290-ca91-4a42-a296-5cd3a560dcea)

3. Looks like styling for the user is a bit off vs design (font weight, size)

![image](https://github.com/hicommonwealth/commonwealth/assets/14819225/25a49109-5c80-44c1-9f6c-1e855d5baccf)

lzach83 commented:
I found nothing more than what @masvelio found, i'll be happy to approve once his comments are addressed

Miaplacidus commented:
> Left some comments and also spotted some things that needs to be improved.
> 
> 1. I am not able to move my mouse over the popover eg if I want to click the user.
> 
>  Kapture.2023-12-14.at.10.25.53.mp4 
> 2. If upvoters array is empty, we display popover with only title. We should not display it at all as we do it currently in upvote button. The problem is that currently we render two different things depending if we have upvoters or not. We should do it better now and handle this logic within the component.
> 
> ![image](https://private-user-images.githubusercontent.com/14819225/290472126-5efb7290-ca91-4a42-a296-5cd3a560dcea.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTEiLCJleHAiOjE3MDI5NDIzNTIsIm5iZiI6MTcwMjk0MjA1MiwicGF0aCI6Ii8xNDgxOTIyNS8yOTA0NzIxMjYtNWVmYjcyOTAtY2E5MS00YTQyLWEyOTYtNWNkM2E1NjBkY2VhLnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFJV05KWUFYNENTVkVINTNBJTJGMjAyMzEyMTglMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjMxMjE4VDIzMjczMlomWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPWZiMjA1ZjM4M2Q5NmQ2Y2U0NmY1YmU4MmVhODg2ZmNkMDA1NmEwYjZlYjgwNTg3Njc5MTg0ZGUzMWJlYzNkOGYmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0JmFjdG9yX2lkPTAma2V5X2lkPTAmcmVwb19pZD0wIn0.v2oThWf61f_i2r4wk85gOXoAAmJt-KEbtI-FDcbTxX0)
> 
> 3. Looks like styling for the user is a bit off vs design (font weight, size)
> 
> ![image](https://private-user-images.githubusercontent.com/14819225/290473041-25a49109-5c80-44c1-9f6c-1e855d5baccf.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTEiLCJleHAiOjE3MDI5NDIzNTIsIm5iZiI6MTcwMjk0MjA1MiwicGF0aCI6Ii8xNDgxOTIyNS8yOTA0NzMwNDEtMjVhNDkxMDktNWM4MC00NGMxLTlmNmMtMWU4NTVkNWJhY2NmLnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFJV05KWUFYNENTVkVINTNBJTJGMjAyMzEyMTglMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjMxMjE4VDIzMjczMlomWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPTU4MmQxODg1Mzg5MTE3NzYyNTkxOGI2NjgxYmFmMTc5NzdlMWRjOWYxNzFjMzAyZTE2YjcwMjNmMjJmMjcyZjcmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0JmFjdG9yX2lkPTAma2V5X2lkPTAmcmVwb19pZD0wIn0.lHKQYaFMzYO_ZBZqfDKoODooFtvEaqbusS85HR9Ww1Y)

Addressed all of these issues. 

Thanks for reviewing, @masvelio!

Miaplacidus commented:
@lzach83 Issues addressed! Please take another look!
